### PR TITLE
Add dtor for WriteBufferToFileSegment

### PR DESCRIPTION
**Pull request by vdimir**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

<details>
    <summary>Modify your CI run</summary>

**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step

#### Include tests (required builds will be added automatically):
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Unit tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with Analyzer
- [ ]  All with Azure
- [ ]  Add your option here

#### Exclude tests:
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with MSAN
- [ ]  All with UBSAN
- [ ]  All with Coverage
- [ ]  All with Aarch64
- [ ]  Add your option here

#### Extra options:
- [ ]  do not test (only style check)
- [ ]  disable merge-commit (no merge from master before tests)
- [ ]  disable CI cache (job reuse)

#### Only specified batches in multi-batch jobs:
- [ ]  1
- [ ]  2
- [ ]  3
- [ ]  4

<details>

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
*This is an automated comment for commit cf28e6a66c43cbb3059742279c2d20facb658ea4 with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/63403/cf28e6a66c43cbb3059742279c2d20facb658ea4/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td><a href="https://github.com/ClickHouse/ClickHouse/blob/master/.github/PULL_REQUEST_TEMPLATE.md?plain=1">❌ failure</a></td></tr>
<tbody>
</table>

**Review by novikd — APPROVED** [review]

File: src/Interpreters/Cache/WriteBufferToFileSegment.h [hunk 1]
```diff
@@ -17,6 +17,8 @@ class WriteBufferToFileSegment : public WriteBufferFromFileDecorator, public IRe
 
     void nextImpl() override;
 
+    ~WriteBufferToFileSegment() override = default;
```
Comment (alexey-milovidov) [inline src/Interpreters/Cache/WriteBufferToFileSegment.h:20]: What does it change?

Comment (vdimir) [reply src/Interpreters/Cache/WriteBufferToFileSegment.h:20]: I assume nothing :) https://github.com/ClickHouse/ClickHouse/pull/63346#discussion_r1589299406

**Comment by alexey-milovidov** [comment]
In fact, using explicit destructors is more confusing than not using them...

**Pull request closed without merging**